### ISSUE TITLE
tag-helper abp-input inside a modal inside a module is not localized

### ISSUE THREAD

**Issue by pjkeukens** [abpframework/abp#12535]
* ABP Commercial Version: 5.2.1
* User Interface: MVC
* Database provider: EF Core

Steps to reproduce:
1. Create new application using abp-suite
2. Add a module to newly created application using abp-suite
3. Add a simple entity ContactPerson (properties name and color) to the MODULE using abp-suite
4. Update the SalesResource (in the module) file de-DE and add "Color": "Kleur" and "Name": "Naam"
5. Generate the migrations and update database
6. run the application and login
7. change language to German
8. go to newly created page ContactPerson (see values of Color and Name are Localized
9. Add new ContactPerson here you see that Name and Color are not Localized

When I add Name and Color to the resource file in the main application then the modals are localized but it should use the localization file in the module.

![contact-person-json](https://user-images.githubusercontent.com/16360696/167822905-68ffb5d1-6828-43b3-ab20-145ae8e5c0fe.JPG)
![contact-person](https://user-images.githubusercontent.com/16360696/167822910-ab6aa6fc-bad3-4953-a1f7-a1e82fe8bd12.JPG)
.

**Comment by yekalkan** [comment]
You need to set the label in html:

`<abp-input asp-for="..." label="L["Color"].Value"/>`

I'll check what we can do to fix this in suite.

**Comment by yekalkan** [comment]
Fixed in the private repository.

**Comment by pjkeukens** [comment]
@yekalkan thanks. How about the abp-input for enum properties, these are not localized as well and you can't add a label there?

**Comment by ebicoglu** [comment]
```
Gender.cs
{
  Unknown = 0,
  Male =1,
  Female = 2
}

```

**en.json**
```
    "Enum:Gender:0": "Unknown",
    "Enum:Gender:1": "Male",
    "Enum:Gender:2": "Female",
```

**Comment by pjkeukens** [comment]
@ebicoglu @yekalkan Yes I have the enum and the translations in my MODULES resource file (it gets added by abp-suite automatically). But it doesn't get localized when you create a new item with the pages that are generated inside the module.

**Comment by yekalkan** [comment]
As i see, the models we use in create/update pages are defined in application layer. (ItemCreateDto & ItemUpdateDto)

We should create ViewModel classes in razor pages.

I'll do it.

**Comment by pjkeukens** [comment]
@yekalkan That would be great but I don't know if it solves the problem.

**Comment by yekalkan** [comment]
It does.

But you may also need to rename the localization keys of enum properties.

From this:

```
    "Enum:Gender:0": "Unknown",
    "Enum:Gender:1": "Male",
    "Enum:Gender:2": "Female",
```

to this: 

````
    "Enum:Gender.Unknown": "Unknown",
    "Enum:Gender.Male": "Male",
    "Enum:Gender.Female": "Female",
````

See https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.AspNetCore.Mvc.UI.Bootstrap/TagHelpers/Form/AbpSelectTagHelperService.cs#L274

**Comment by pjkeukens** [comment]
@yekalkan yesterday I moved some of the pages out of the module and placed them in the main application. The select boxes still weren't localized. I added both notations above in my resource file but still no luck.

Then I saw that one of the three items was localized and the other two weren't. I was baffled so I thought why would that one be localized and not the others. I found out that it was translated because the enum name was in the resource file. So to follow your example "Enum:Gender:0": "Unknown" and "Enum:Gender.Unknown": "Unknown" weren't localized but "Unknown": "Unknown" would localize the enum value in the select box. any ideas?

so I added an underscore at the end of the enum name and added them to the resource file and all were localized... It's a bit weird.
